### ISSUE TITLE
Very slow to recognize marker

### ISSUE THREAD

**Issue by raddevon** [viromedia/viro#318]
I've got this really simple scene:

```
"use strict";

import React, { Component } from "react";

import {
  ViroARScene,
  ViroARImageMarker,
  ViroARTrackingTargets,
  ViroAmbientLight,
  Viro3DObject
} from "react-viro";

export default class HelloWorldSceneAR extends Component {
  state = {
    pauseUpdates: false
  };

  render() {
    return (
      <ViroARScene>
        <ViroARImageMarker
          target={"test"}
          onAnchorFound={this._onAnchorFound}
          pauseUpdates={this.state.pauseUpdates}
        >
          <ViroAmbientLight color="#ffffff" />
          <Viro3DObject
            source={require("./res/emoji_smile/emoji_smile.vrx")}
            position={[0, 0.01, 0]}
            scale={[0.5, 0.5, 0.5]}
            rotation={[-90, 0, 0]}
            type="VRX"
          />
        </ViroARImageMarker>
      </ViroARScene>
    );
  }

  _onAnchorFound = () => {
    this.setState({ pauseUpdates: true });
  };
}

ViroARTrackingTargets.createTargets({
  test: {
    source: require("./res/test.png"),
    orientation: "Up",
    physicalWidth: 0.18
  }
});

module.exports = HelloWorldSceneAR;
```

It takes a long time to get the 3D object to display, as if the marker is not being recognized. It takes sometimes 45 seconds to a minute of walking around with the marker in frame, viewing it from different angles and distances before the object finally pops into view.

target image:
<img width="306" alt="test" src="https://user-images.githubusercontent.com/2027711/41206462-33f2ddb4-6cb9-11e8-97c4-b500c0947e31.png">

I printed this image out on a sheet of paper to use at the marker.

I don't understand if this is some problem with my code, the marker image, my lighting my device... it definitely feels like it shouldn't take this long, but I don't understand the problem. Can anyone help me figure this out?

**Comment by saad-ansari** [comment]
i'm also facing the same issue even the camera quality is very bad and lags a lot.

**Comment by raddevon** [comment]
@dam00n I'm on an iPhone 7 running iOS 11.3.1. I am on ViroReact 2.7.3. I should also note that, unlike @saad-ansari, my camera image is fast and smooth. It's just the marker detection that takes a long time.

I was afraid the marker image might be my issue. Is there any documentation on how to create a good marker image?

I'll try the Tesla marker demo and report back. Thank you!

**Comment by dam00n** [comment]
Here is Apple's guidance -> https://developer.apple.com/design/human-interface-guidelines/ios/system-capabilities/augmented-reality/

"Design and display reference images to optimize detection. When you provide reference images, you specify the physical size at which you expect to find those images in the user’s environment. Providing a more precise size measurement helps ARKit detect images faster and provide more accurate estimates of their real-world position. Detection performance and accuracy are best for flat rectangular images with high contrast and bold details. Avoid trying to detect images that appear on reflective or curved real-world surfaces."

While not the most specific guidance, "high contrast and bold details" has been the most applicable in our testing. Examples include movie posters, magazine covers, paintings, etc.

**Comment by ebradbury** [comment]
I'm experiencing the same issue on an iPhone SE (11.4). I have a very similar marker image that bold black text on a white background. I double checked my dimensions but the marker is rarely detected.

Should the content inside of my ViroARImageMarker automatically be oriented onto the plane of the marker? My ViroBox inside of the ViroARImageMarker is appearing elsewhere in the world, at odd orientations.

**Comment by saad-ansari** [comment]
@dam00n i'm using Samsung S7.

**Comment by dam00n** [comment]
@ebradbury can you try another marker? Would like to see if it is an issue with your specific marker or a device issue. Here is a marker example you can try -> https://github.com/viromedia/viro/tree/master/js/ARCarDemo

If this is an issue with the marker, you might need to add more contract and bold details to the image marker

Yes, the content should be automatically oriented onto the plane of the marker. You may need to rotate your object 90 degrees depending on if you want it match the image or sit on top of the image. Can you try with a different marker and see if you are seeing the same issues?

**Comment by dam00n** [comment]
@saad-ansari can you try restarting your device? We have multiple S7 we test on and are not seeing the same issue. If that does not do anything, can you do the following:

1. Create a new github issue as the poor quality camera is a separate issue
2. Provide a screenshot or video of the issue
3. device is S7 but what OS version are you running? And what version of ViroReact and ARCore are you running?

**Comment by raddevon** [comment]
@dam00n Just got a chance to try this. The Tesla demo is super-fast to recognize the marker on my device. I guess this means my markers are most likely the reason for the slow response. Is that your assumption here as well?

**Comment by dam00n** [comment]
@raddevon yep that is our assumption also. Perhaps try adding some color or adding an icon to your marker?

**Comment by ebradbury** [comment]
@dam00n I changed my marker to have color and it's detecting it, although not tracking it particularly well. The ViroBox shows up but drifts away from the marker as I move my device around. I'll try the example and see if it makes a difference. I was expecting the augmented content to "stick" tightly to the marker once it was detected.

**Comment by dam00n** [comment]
@ebradbury That is interesting. Try the example and let us know if you still see the drift. That way we can tell if it is your marker or the platform.

**Comment by yashank09** [comment]
Hey @dam00n ! 
I'm facing the same issue where the markers get detected but dont 'stick' to the object. It might be due to multiple markers or marker positions, will be looking more in the matter.
